File: tutorials/testing/c++/tut_us_map_p3.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/USState.h"
#include "data_src/USCounty.h"
#include "USMap.h"

using namespace bridges;

// This program illustrates how to access the data of the US map with state
// boundaries with different colors for states and its boundaries
// Here we select 3 states and draw them with different attributes with or 
// without counties
int main(int argc, char **argv) {

	// create bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	// set title
	bridges.setTitle("Tutorial: Drawing US Map with States and Counties");
	bridges.setDescription("3 US states, with different attributes");

	// create a data source object
	DataSource ds(&bridges);
	
	// specify the states of interest
	vector<string> states = {"Vermont", "Iowa", "North Carolina"};
	
	// get us map data - all states and counties
	// we also retrieve the counties, using the second argument
	vector<bridges::dataset::USState> map_data = ds.getUSMapCountyData (states, true); 

	// we will draw the first state without counties and set fill and boundary 
	// colors - use the viewCounties() flag to ignore counties
	auto& s = map_data[0];
	s.setStrokeColor(Color(255,0,0));
	s.setFillColor(Color(0,255,0));
	s.setViewCountiesFlag(false);

	// the second state  we will draw with counties and set their colors
	auto& s2 = map_data[1];
	// iterate over the counties
	for (auto& c: s2.accessCounties()) {
		c.second.setStrokeColor(Color(50,250,50));
		c.second.setFillColor(Color(0,0,25));
	}

	// the third state will use default colors

	// create a USMap object with the map data
	USMap us_maps(map_data);

	// set the data structure
	bridges.setDataStructure(&us_maps);

	//visualize
	bridges.visualize();
	
	return 0;
}

File: tutorials/testing/python/tut_usmap_p3.py
from bridges.bridges import *
from bridges.data_src_dependent import *
from bridges.dl_element import *
from bridges.us_map import *

def main():
    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # set title, description
    bridges.set_title("Tutorial: Drawing US Map with States and Counties")
    bridges.set_description("3 US  states, with or without Counties, with varied coloring")

    #pass no parameters for the whole map
    #state_info = get_us_map_county_data(['Vermont', 'Iowa', 'North Carolina'], True)
    state_info = get_us_map_county_data(['Vermont', 'Iowa', 'North Carolina'], False)

    # we will draw the first state without counties and set 
    # fill and boundary colors - use the viewCounties() flag to 
    # ignore counties
    state_info[0].view_counties = False
    state_info[0].stroke_color = Color(255,0,0)
    state_info[0].fill_color = Color(0,255,0)

    # the second state  we will draw with counties and set their colors
    for c in state_info[1].counties:
        c.stroke_color = Color(50, 250, 50)
        c.fill_color = Color(0,0,170)

    # the third state will use default colors
    
        
    my_map = USMap(state_info)
    
    #visualizes iowa and california with counties
    bridges.set_data_structure(my_map)
    bridges.visualize()


if __name__ == '__main__':
    main()
